fastai/data_block.py

```diff
@@ -709,16 +709,21 @@ def transform(self, tfms:TfmList, tfm_y:bool=None, **kwargs):
         _check_kwargs(self.x, tfms, **kwargs)
         if tfm_y is None: tfm_y = self.tfm_y
         if tfm_y: _check_kwargs(self.y, tfms, **kwargs)
-        self.tfms,self.tfmargs = tfms,kwargs
-        self.tfm_y,self.tfms_y,self.tfmargs_y = tfm_y,tfms,kwargs
+        self.tfms,  self.tfmargs   = tfms,kwargs
+        self.tfm_y, self.tfmargs_y = tfm_y,kwargs
+        self.tfms_y = None if tfms is None else list(filter(lambda t: t.use_on_y, tfms))
         return self
 
     def transform_y(self, tfms:TfmList=None, **kwargs):
         "Set `tfms` to be applied to the targets only."
         _check_kwargs(self.y, tfms, **kwargs)
         self.tfm_y=True
-        if tfms is None: self.tfms_y,self.tfmargs_y = self.tfms,{**self.tfmargs, **kwargs}
-        else:            self.tfms_y,self.tfmargs_y = tfms,kwargs
+        if tfms is None:
+            self.tfms = list(filter(lambda t: t.use_on_y, self.tfms))
```

Comment:
Why change `self.tfms`? It's only `self.tfms_y` that need to be the filtered list, no?

Reply:
yep, you're right